src/api.v2/model/Sample.js

```diff
@@ -16,6 +16,47 @@ class Sample extends BasicModel {
   constructor(sql = sqlClient.get()) {
     super(sql, tableNames.SAMPLE, sampleFields);
   }
+
+  async getSamples(experimentId) {
+    const { sql } = this;
+
+    const fieldsWithMetadata = [...sampleFields, sql.raw(
+      `${replaceNullsWithObject('jsonb_object_agg(key, value)', 'key')} as metadata`,
+    )];
+    const sampleFieldsWithAlias = sampleFields.map((field) => `s.${field}`);
+
+    const metadataQuery = sql.select(fieldsWithMetadata)
+      .from(sql.select([...sampleFieldsWithAlias, 'm.key', 'sm_map.value'])
+        .from({ s: tableNames.SAMPLE })
+        .leftJoin(`${tableNames.METADATA_TRACK} as m`, 's.experiment_id', 'm.experiment_id')
+        .leftJoin(`${tableNames.SAMPLE_IN_METADATA_TRACK_MAP} as sm_map`, 's.id', 'sm_map.sample_id')
+        .where('s.experiment_id', experimentId)
+        .as('mainQuery'))
+      .groupBy(sampleFields)
+      .as('select_metadata');
+
+    const sampleFileFields = ['sample_file_type', 's3_path', 'size', 'valid', 'upload_status'];
+    const sampleFileFieldsWithAlias = sampleFileFields.map((field) => `sf.${field}`);
+    const fileObjectFormatted = sampleFileFields.map((field) => [`'${field}'`, field]);
+
+    const fileNamesQuery = sql.select(['id', sql.raw(
+      `jsonb_object_agg(sample_file_type,json_build_object(${fileObjectFormatted})) as files`,
```

Comment:
Similar to the suggestion for metadata track, we could write this instead:

`const sampleFileObject = `jsonb_object_agg(sample_file_type,json_build_object(${fileObjectFormatted})) as files``

and then it would be:
`sql.select(['id', sql.raw(sampleFilesObject)])`